### PR TITLE
[Android] Added feature, switch between  speakers and earpiece

### PR DESCRIPTION
**Pull request by beerline**
# Description
Feature to toggle between speakers and earpiece. Can be used to play some thing when no headphones nearby and for not disturbing people around. Switch to headphones if connected (wire and wireless).

# Affected methods and callbacks:
1. Added `Content` into a methods `play`, `setUrl` of class `Player`
2. Added `Content` into `setAttributes` of class `WrappedMediaPlayer`
3. Added condition to choose between speakers and earpiece into `setAttributes` of class `WrappedMediaPlayer`
4. Added method `setPlayingRoute` into class `Player`
5. Added implementation of `setPlayingRoute` into `WrappedMediaPlayer`
6. Added new method `handleMethodCall` in to `AudioplayersPlugin` class
7. Added example

# Supporting platforms 
Implemented only for Android

# Tested
- Android v.10 google pixel
- Android v.10 API 29 emulator
- Android v.4.3.1 API 18 emulator

### CODE REVIEW

File: lib/audioplayers.dart [hunk 1]
```diff
@@ -602,4 +614,24 @@ class AudioPlayer {
 
     await Future.wait(futures);
   }
+
+  Future<int> earpieceOrSpeakersToggle() async {
+    PlayingRouteState playingRoute = _playingRouteState == PlayingRouteState.EARPIECE
+        ? PlayingRouteState.SPEAKERS
+        : PlayingRouteState.EARPIECE;
+
+//    return _invokeMethod('setVolume', {'volume': volume});
```
Comment (luanpotter) [inline lib/audioplayers.dart]: please remove commented out code

File: android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java [hunk 2]
```diff
@@ -258,16 +291,35 @@ private void setSource(String url) {
...
+                if ( context != null ) {
+                    AudioManager mAudioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
+                    mAudioManager.setSpeakerphoneOn(false);
+                }
+            }
+
         } else {
             // This method is deprecated but must be used on older devices
-            player.setAudioStreamType(respectSilence ? AudioManager.STREAM_RING : AudioManager.STREAM_MUSIC);
+            if ( objectEquals(this.playingRoute, "speakers") ) {
```
Comment (luanpotter) [inline android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java]: please remove these spaces right after and before the parenthesis:
```
if (objectEquals(a, b)) {
}
```

File: android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java [hunk 3]
```diff
@@ -258,16 +291,35 @@ private void setSource(String url) {
     }
 
     @SuppressWarnings("deprecation")
-    private void setAttributes(MediaPlayer player) {
+    private void setAttributes(MediaPlayer player, Context context) {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-            player.setAudioAttributes(new AudioAttributes.Builder()
+            if ( objectEquals(this.playingRoute, "speakers") ) {
```
Comment (luanpotter) [inline android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java]: please remove these spaces right after and before the parenthesis

File: android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java [hunk 4]
```diff
@@ -67,6 +68,38 @@ void setVolume(double volume) {
...
+
+            int position = 0;
+            if (player != null) {
+                position = player.getCurrentPosition();
+            }
+
+            this.released = false;
+            this.player = createPlayer(context);
+            this.setSource(url);
+            try{
```
Comment (luanpotter) [inline android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java]: `try {`

File: android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java [hunk 5]
```diff
@@ -67,6 +68,38 @@ void setVolume(double volume) {
         }
     }
 
+    @Override
+    void setPlayingRoute(String playingRoute, Context context) {
+        if ( ! objectEquals(this.playingRoute, playingRoute) ) {
```
Comment (luanpotter) [inline android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java]: please remove these spaces right after and before the parenthesis:
```
if (!objectEquals(a, b)) {
}
```

File: README.md [hunk 6]
```diff
@@ -280,6 +287,7 @@ Here is an example of how it should look like:
     </application>
 </manifest>
 ```
+## :warning: iOS stream routing not implemented
```
Comment (luanpotter) [inline README.md]: should this warning be present on the `Stream routing` subsection as well?

**Comment by beerline** [comment]
Thank you for code review! 
I fixed all formatting issues. 

Also i will be appreciate, if some one helps with iOS implementation of stream routing.

**Comment by luanpotter** [comment]
@beerline it would be great, but I think we can merge this one and get the functionality for android. can you open an issue (if doesn't already exist) requesting this to be impl for ios as well? Hopefully someone with ios expertise can help us out.

**Pull request merged**